### PR TITLE
Added building condition esp

### PR DESCRIPTION
**Pull request by NotAnUser1**
yeah thats it ig

### CODE REVIEW

**Comment by TotallyNotElite** [comment]
I'll review this soont :tm:, please don't merge without my approval

File: src/hacks/ESP.cpp [hunk 1]
```diff
@@ -1294,31 +1296,85 @@ void _FASTCALL ProcessEntity(CachedEntity *ent)
...
+                    }
+                }
+                break;
+            }
+            }
+
+            if (IsSapped)
+                AddEntityString(ent, sapped_str, colors::FromRGBA8(220.0f, 220.0f, 220.0f, 255.0f));
+
+            else if (classid == CL_CLASS(CObjectTeleporter) && CE_INT(ent, netvar.m_iTeleState) == 1 || (IsDisabled || IsPlasmaDisabled))
```
Comment (BenCat07) [inline src/hacks/ESP.cpp:1374]: this should check state <= 1 (0 is disabled, 1 is idle aka unconnected)

File: src/hacks/ESP.cpp [hunk 2]
```diff
@@ -1294,31 +1296,93 @@ void _FASTCALL ProcessEntity(CachedEntity *ent)
...
+                    break;
+                }
+
+                AddEntityString(ent, format(std::to_string(sentry_ammo), '/', max_ammo, " Ammo"));
+
+                if (CE_INT(ent, netvar.iUpgradeLevel) == 3)
+                    AddEntityString(ent, format(std::to_string(sentry_rockets), '/', "20 Rockets"));
+
+                if (IsControlled) // Dispensers are also "controlled" when they're in use
+                    AddEntityString(ent, controlled_str, colors::FromRGBA8(220.0f, 220.0f, 220.0f, 255.0f));
```
Comment (TotallyNotElite) [inline src/hacks/ESP.cpp:1349]: I don't think we should show this at all actually. You don't seem to use IsControlled anywhere else, so you should also remove the bool too.

```suggestion
```

File: src/hacks/ESP.cpp [hunk 3]
```diff
@@ -1294,31 +1296,93 @@ void _FASTCALL ProcessEntity(CachedEntity *ent)
...
+            bool IsDisabled       = CE_BYTE(ent, netvar.m_bDisabled);
+            bool IsPlasmaDisabled = CE_BYTE(ent, netvar.m_bPlasmaDisable);
+            bool IsMini           = CE_BYTE(ent, netvar.m_bMiniBuilding);
+            int required_metal    = CE_INT(ent, netvar.m_iUpgradeMetalRequired);
+            int metal             = CE_INT(ent, netvar.m_iUpgradeMetal);
+            
+            if (!IsMini && CE_INT(ent, netvar.iUpgradeLevel) != 3)
+                AddEntityString(ent, format("Upgrade: " + std::to_string(required_metal - metal), '/', std::to_string(required_metal)));
+
+            switch (classid)
```
Comment (TotallyNotElite) [inline src/hacks/ESP.cpp:1323]: Can you remove the useless curly brackets after the `case` statements?

File: include/core/netvars.hpp [hunk 4]
```diff
@@ -1,4 +1,4 @@
-﻿/*
+/*
```
Comment (TotallyNotElite) [inline include/core/netvars.hpp:1]: What even changed here?

**Review by TotallyNotElite — CHANGES_REQUESTED** [review]

File: src/hacks/ESP.cpp (hunk 2 above)
Comment (NotAnUser1) [reply src/hacks/ESP.cpp:1349]: Sure, I can remove the bool but can I keep this in? imo it's pretty informative

Comment (BenCat07) [reply src/hacks/ESP.cpp:1348]: yes

**Review by TotallyNotElite — APPROVED** [review]
I'm trusting my previous review here, LGTM.

**Pull request merged**